Source freshness (#6744)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/source.md

If your dbt project is [configured with sources](/docs/build/sources), then the `dbt source freshness` command will query all of your defined source tables, determining the "freshness" of these tables. If the tables are stale (based on the `freshness` config specified for your sources) then dbt will report a warning or error accordingly. If a source <Term id="table" /> is in a stale state, then dbt will exit with a nonzero exit code.

You can also use [source freshness](/docs/deploy/source-freshness) commands help make sure the data you get is new and not old or outdated. 

#### Configure source freshness

You can configure source freshness in the execution settings within your job in dbt Cloud. For more information, refer to [enabling source freshness snapshots](/docs/deploy/source-freshness#enabling-source-freshness-snapshots).

> import SourcefreshnessTable from '/snippets/_selecting_source_freshness.md';

Reviewer: it looks like we're using this snippet in one place only? maybe for this scenario, it's easier to just use the image? 

snippets are useful if you want to display content in multiple pages

If your dbt project is [configured with sources](/docs/build/sources), then the `dbt source freshness` command will query all of your defined source tables, determining the "freshness" of these tables. If the tables are stale (based on the `freshness` config specified for your sources) then dbt will report a warning or error accordingly. If a source <Term id="table" /> is in a stale state, then dbt will exit with a nonzero exit code.

> You can also use [source freshness](/docs/deploy/source-freshness) commands help make sure the data you get is new and not old or outdated.

Author: ```suggestion
You can also use [source freshness commands](/reference/commands/source#source-freshness-commands) to help make sure the data you get is new and not old or outdated.
```

### Configure source freshness

The example below, shows how to configure source freshness in dbt. Refer to [Declaring source freshness](/docs/build/sources#declaring-source-freshness) for more information. 

<File name='models/<filename>.yml'>

```yaml

version: 2

sources:
  - name: jaffle_shop
    database: raw

    freshness:
      warn_after: {count: 12, period: hour}
      error_after: {count: 24, period: hour}

    loaded_at_field: _etl_loaded_at

    tables:
      - name: customers

      - name: orders
        freshness:
          warn_after: {count: 6, period: hour}
          error_after: {count: 12, period: hour}
          filter: datediff('day', _etl_loaded_at, current_timestamp) < 2

      - name: product_skus
        freshness: null

```
</File>

This helps to monitor the data pipeline health.

> You can also configure source freshness in the execution settings within your job in dbt Cloud. For more information, refer to [Enabling source freshness snapshots](/docs/deploy/source-freshness#enabling-source-freshness-snapshots).

Reviewer: non blocker but would be nice to have an image? but maybe not necessary
```suggestion
You can also configure source freshness in the **Execution settings** section in your dbt Cloud job **Settings** page. For more information, refer to [Enabling source freshness snapshots](/docs/deploy/source-freshness#enabling-source-freshness-snapshots).
```

If your dbt project is [configured with sources](/docs/build/sources), then the `dbt source freshness` command will query all of your defined source tables, determining the "freshness" of these tables. If the tables are stale (based on the `freshness` config specified for your sources) then dbt will report a warning or error accordingly. If a source <Term id="table" /> is in a stale state, then dbt will exit with a nonzero exit code.

> Refer to [Declaring source freshness](/docs/build/sources#declaring-source-freshness) for more information. 

Author: ```suggestion
You can also use [source freshness](/docs/deploy/source-freshness) commands help make sure the data you get is new and not old or outdated.
```

> You can also use [source freshness](/docs/deploy/source-freshness) commands help make sure the data you get is new and not old or outdated. 

Author: ```suggestion
Refer to [Declaring source freshness](/docs/build/sources#declaring-source-freshness) for more information. 
```

> ### Configure source freshness

Reviewer: very cool! this seems a little duplicative/repetitive of [this page though](https://docs-getdbt-com-git-nfiann-sourcefreshness-dbt-labs.vercel.app/docs/build/sources#declaring-source-freshness) -- should we not just link to docs/build/sources page here?

          error_after: {count: 12, period: hour}
          filter: datediff('day', _etl_loaded_at, current_timestamp) < 2

      - name: product_skus
        freshness: null

```
</File>

This helps to monitor the data pipeline health.

You can also configure source freshness in the execution settings within your job in dbt Cloud. For more information, refer to [Enabling source freshness snapshots](/docs/deploy/source-freshness#enabling-source-freshness-snapshots).

### Source freshness commands

Source freshness commands ensure you're receiving the most up-to-date, relevant, and accurate information. 

Some of the typical commands you can use are:

| **Command**                                                                 | **Description**                  | 
| ----------------------------------------------------------------------------| ---------------------------------|
|[`dbt source freshness`](/reference/commands/source#dbt-source-freshness)    |Checks the "freshness" for all sources.|
|[`dbt source freshness --output target/source_freshness.json`](/reference/commands/source#configuring-source-freshness-output)|Output of "freshness" information to a different path.|

> |[`dbt source freshness --select "source:snowplow"`](/reference/commands/source#specifying-sources-to-snapshot)|Checks the "freshness" for specific sources.|

Reviewer: i think the snowplow here is just the source name. would it be better to have a placeholder so users don't think they have to write snowplow? e.g. `dbt source freshness --select "source:source_name"`

> |[`dbt source freshness --select "source:snowplow"`](/reference/commands/source#specifying-sources-to-snapshot)|Checks the "freshness" for specific sources.|

Author: ```suggestion
|[`dbt source freshness --select "source:source_name"`](/reference/commands/source#specifying-sources-to-snapshot)|Checks the "freshness" for specific sources.|
```

This command can be run manually to determine the state of your source data freshness at any time. It is also recommended that you run this command on a schedule, storing the results of the freshness snapshot at regular intervals. These longitudinal snapshots will make it possible to be alerted when source data freshness SLAs are violated, as well as understand the trend of freshness over time.

dbt Cloud makes it easy to snapshot source freshness on a schedule, and provides a dashboard out of the box indicating the state of freshness for all of the sources defined in your project. For more information on snapshotting freshness in dbt Cloud, check out the [docs](/docs/build/sources#source-data-freshness).

### Source freshness commands

Source freshness commands ensure you're receiving the most up-to-date, relevant, and accurate information. 

Some of the typical commands you can use are:

| **Command**                                                                 | **Description**                  | 
| ----------------------------------------------------------------------------| ---------------------------------|

> |[`source_status`](/reference/node-selection/methods#the-source_status-method)|dbt generates the [`sources.json`](/reference/artifacts/sources-json) artifact, which includes execution times and [`max_loaded_at`](/reference/artifacts/sources-json#top-level-keys) timestamps for dbt sources.|

Reviewer: ```suggestion
|[`source_status`](/reference/node-selection/methods#source_status) | dbt generates the [`sources.json`](/reference/artifacts/sources-json) artifact, which includes execution times and [`max_loaded_at`](/reference/artifacts/sources-json#top-level-keys) timestamps for dbt sources.|
```

Author: Hiya @mirnawong1 

Apologies for reviewing your review (not sure if we're allowed to do that) but the link: reference/node-selection/methods#source_statusd takes me to the Node Selector page. Would it be ok if I update the link to: reference/node-selection/methods#source_status for the source status section?

Kind Regards
Natalie

Reviewer: yes that's fine! my bad, must have had a typo for typing too fast! the original link you had `/reference/node-selection/methods#the-source_status-method` didn't take me to the right place
